src/SDKs/Search/DataPlane/Microsoft.Azure.Search/Properties/AssemblyInfo.cs

```diff
@@ -1,11 +0,0 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License. See License.txt in the project root for
-// license information.
-
-using System.Reflection;
-
-[assembly: AssemblyVersion("9.0.0.0")]
```

Comment:
@brjohnstmsft do you guys intentionally keep the AssemblyVersion fixed? By default when we start letting the SDK generate these it will generate them based on the Version in the project. However if you want to keep it fixed you can set another property to do that.

Reply:
@weshaggard Our goal is to follow the SemVer rules as they apply to .NET, which means that the AssemblyVersion's major component should match the major version of the NuGet package, and the rest (minor, patch, and build numbers) should all be zero. Then to make the assemblies self-describing, we want the AssemblyFileVersion to reflect exactly the NuGet package version.

I'm not opinionated about how that policy is implemented, as long as it works consistently and predictably.

Reply:
I'm not aware of any SemVer rules that state that we should keep the assembly version parts to 0. That actually might cause problems if you add new APIs and people try to resolve your assembly in an application when 2 libraries use your libraries with the same assembly version but one depends on newer APIs then the other. I suggest we keep the assembly version, file version, and package version consistent unless there is a specific reason you need to make them differ.

Reply:
We never make breaking changes without bumping the major version number, so the scenario you describe shouldn't happen. On the other hand, if we always force everyone to bind to a specific version of our assemblies, it makes dependency management very brittle since even a tiny point-release with non-breaking bug fixes would be treated as a completely different assembly (due to having a different strong name). So a customer using two libraries that each in turn take dependencies on different minor or patch versions of our assemblies would be out of luck. Am I not understanding that correctly?

Reply:
It isn't about breaking changes. If you even add new APIs across minor or patch versions then you have this potential problem. I'm happy to chat and walk you through the those potential issues if you want to better understand. 

>a customer using two libraries that each in turn take dependencies on different minor or patch versions of our assemblies would be out of luck. 

They aren't out of luck they can correctly unify via binding redirects on .NET framework and just version in .NET Core. In fact if you you don't bump the version there is a good chance they may be out of luck because they might not have the fixes or new apis they expect or depend on.

Reply:
I don't have a lot of experience with .NET Core, but binding redirects in .NET Framework have been the bane of my existence. If I understand correctly, that's all handled automatically in .NET Core, so now you can just do this:

```xml
<PropertyGroup>
  <AutoGenerateBindingRedirects>true</AutoGenerateBindingRedirects>
  <GenerateBindingRedirectsOutputType>true</GenerateBindingRedirectsOutputType>
</PropertyGroup>
```

Is that right?

If this is the state of the art now, then I'm all in favor of not doing things differently. Go ahead and set our versions consistently with other Azure SDKs.

Reply:
.NET Core doesn't use binding redirects it always just unifies to the highest loaded version. For .NET Framework the AutoGenerateBindingRedirects is a common way for applications to resolve the binding redirect problem and for newer tools it is on by default (https://github.com/dotnet/sdk/issues/1405). 

So my recommendation would be to use the same version for consistency and if we discover issues we can reconsider options.

Reply:
As long as that's the standard practice, I'm fine with it. Thanks!